app/controllers/task_contributions_controller.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+class TaskContributionsController < ApplicationController
+  include TaskParameters
+
+  before_action :load_and_authorize_base_task
+  before_action :load_and_authorize_task_contribution, except: %i[index create new]
+
+  def approve_changes
+    if @task.apply_contribution(@task_contribution)
+      TaskContributionMailer.send_approval_info(@task_contribution).deliver_later
+      redirect_to @task, notice: t('.success')
+    else
+      redirect_to [@task, @task_contribution], alert: t('.error')
+    end
+  end
+
+  def discard_changes
+    duplicate = @task_contribution.decouple
+    if duplicate
+      redirect_to duplicate, notice: t('.success')
+    else
+      redirect_to [@task, @task_contribution], alert: t('.error')
+    end
+  end
+
+  def reject_changes
+    duplicate = @task_contribution.decouple
+    if duplicate
+      TaskContributionMailer.send_rejection_info(@task_contribution, duplicate).deliver_later
+      redirect_to @task, notice: t('.success')
+    else
+      redirect_to [@task, @task_contribution], alert: t('.error')
+    end
+  end
+
+  def index
+    @task_contributions = @task.contributions.order(created_at: :desc).paginate(page: params[:page], per_page: per_page_param)
+    raise Pundit::NotAuthorizedError unless policy(TaskContribution).index? base: @task
```

Comment:
```suggestion
    raise Pundit::NotAuthorizedError unless policy(@task_contributions).index? base: @task
```

This change will fail the `spec/controllers/task_contributions_controller_spec.rb` for `index`, which will be fixed by adding `let(:user) { original_author }` there.

Reply:
I have implemented the changes.